src/api/app/models/workflow/step.rb

```diff
@@ -37,16 +37,26 @@ def create_or_update_subscriptions(package, workflow_filters)
 
   protected
 
-  def validate_step_instructions
+  def validate_step_keys
```

Comment:
I would keep the validations in a single method to not loop twice over the required keys. It's not much overheard, but the code could be like this:

```ruby
  def validate_step_instructions
    self.class::REQUIRED_KEYS.each do |required_key|
      unless step_instructions.key?(required_key)
        errors.add(:base, "The '#{required_key}' key is missing")
        next
      end

      errors.add(:base, "The '#{required_key}' key must provide a value") if step_instructions[required_key].blank?
    end
  end
```

Reply:
:+1: done